test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1430 @@
-            'policy': 'ACCEPT',
-            'chain': 'INPUT',
-            '_ansible_check_mode': True,
-        })
-        commands_results = [
-            (0, 'Chain INPUT (policy DROP)\n', ''),
-        ]
-
-        with patch.object(basic.AnsibleModule, 'run_command') as run_command:
-            run_command.side_effect = commands_results
-            with self.assertRaises(AnsibleExitJson) as result:
-                iptables.main()
-                self.assertTrue(result.exception.args[0]['changed'])
-
-        self.assertEqual(run_command.call_count, 1)
-        self.assertEqual(run_command.call_args_list[0][0][0], [
-            '/sbin/iptables',
-            '-t',
-            'filter',
-            '-L',
-            'INPUT',
-        ])
+
+IPTABLES_CMD = "/sbin/iptables"
+IPTABLES_VERSION = "1.8.2"
+CONST_INPUT_FILTER = [IPTABLES_CMD, "-t", "filter", "-L", "INPUT",]
+
+
+@pytest.fixture
+def _mock_basic_commands(mocker):
+    """Mock basic commands like get_bin_path and get_iptables_version."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.get_bin_path",
+        return_value=IPTABLES_CMD,
+    )
+    mocker.patch("ansible.modules.iptables.get_iptables_version", return_value=IPTABLES_VERSION)
+
+
+def test_without_required_parameters(mocker):
+    """Test module without any parameters."""
+    mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.fail_json",
+        side_effect=fail_json,
+    )
+    set_module_args({})
+    with pytest.raises(AnsibleFailJson) as exc:
+        iptables.main()
+
+    assert exc.value.args[0]["failed"]
+    assert "Failed to find required executable" in exc.value.args[0]["msg"]
+
+
+@pytest.mark.usefixtures('_mock_basic_commands')
+def test_flush_table_without_chain(mocker):
+    """Test flush without chain, flush the table."""
+    set_module_args(
+        {
+            "flush": True,
+        }
+    )
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command", return_value=(0, "", "")
+    )
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args[0][0]
+    assert first_cmd_args_list[0], IPTABLES_CMD
+    assert first_cmd_args_list[1], "-t"
+    assert first_cmd_args_list[2], "filter"
+    assert first_cmd_args_list[3], "-F"
+
+
+@pytest.mark.usefixtures('_mock_basic_commands')
+def test_flush_table_check_true(mocker):
+    """Test flush without parameters and check == true."""
+    set_module_args(
+        {
+            "flush": True,
+            "_ansible_check_mode": True,
+        }
+    )
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command", return_value=(0, "", "")
+    )
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 0
+
+
+@pytest.mark.usefixtures('_mock_basic_commands')
+def test_policy_table(mocker):
```

Comment:
@Akasurde this test ended up shadowed by the function definition with the same name `test_policy_table` below. So it doesn't run anymore. Could you fix this?

https://app.codecov.io/gh/ansible/ansible/commit/8750a1f9ace81f57e33a9e3f0234f8f6c1c76e66#a184f210bf607737e23e4930d925ba5f-R82